docker-compose.yml

```diff
@@ -0,0 +1,42 @@
+version: '3.8'
+
+services:
+  mysql:
+    image: mysql:9.0.0
+    container_name: mysql-cont
+    ports:
+      - "3307:3307"
+    environment:
+      MYSQL_ROOT_PASSWORD: root
+      MYSQL_DATABASE: usersystem
+      MYSQL_PASSWORD: root
+      MYSQL_TCP_PORT: 3307
+    volumes:
+      - mysql-data:/var/lib/mysql
+    networks:
+      - app-network
+    restart: always
+
+  spring-boot-app:
+    build: .
+    container_name: spring-boot-app
+    ports:
+      - "8080:8080"
+    environment:
+      SPRING_DATASOURCE_URL: jdbc:mysql://mysql:3307/usersystem?useSSL=false&serverTimezone=UTC&createDatabaseIfNotExist=true&allowPublicKeyRetrieval=true
+      SPRING_DATASOURCE_USERNAME: root
+      SPRING_DATASOURCE_PASSWORD: root
```

Comment:
Defining a password here might be fine, since MySQL container runs locally with dummy data. However, in a docker-compose files, the DB URL might be pointing to a cloud database and its better to pick up the password from a .env file (In any case as a best practice) so it doesn't appear on any public repo.
Here we can include a comment, telling the user to define a .env file with their choice of values and put that in .gitignore